homeassistant/components/fritz/config_flow.py

```diff
@@ -244,3 +267,54 @@ async def async_step_import(self, import_config):
                 CONF_PORT: import_config.get(CONF_PORT, DEFAULT_PORT),
             }
         )
+
+
+class FritzBoxToolsOptionsHandler(OptionsFlow):
+    """Handle a option flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle options flow."""
+        fritz_tools: FritzBoxTools = self.hass.data[DOMAIN][self.config_entry.entry_id][
+            FRITZ_TOOLS
+        ]
+        devices: dict[str, str] = {
+            mac: fritz_device_name(device)
+            for mac, device in fritz_tools.devices.items()
+        }
+
+        if user_input is not None:
+            if user_input[CONF_ADD_ALL_TRACKER]:
+                user_input[CONF_ADD_NEW_TRACKER] = True
+                user_input[CONF_SELECTED_DEVICES] = list(devices.keys())
+                pass
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_ADD_ALL_TRACKER,
```

Comment:
Where is this used?

Reply:
It is also used on configuration options, in case a user want to add all devices to be tracked.

Reply:
But where is it used in the integration?

Reply:
when `CONF_ADD_ALL_TRACKER` is set `True` than `CONF_ADD_NEW_TRACKER` will also set to `True` and all devices will be selected here:
https://github.com/home-assistant/core/blob/3d690485e4e5b8b3eea0996a540a79fb8ee82c89/homeassistant/components/fritz/config_flow.py#L291-L294

Reply:
So if we don't add the other options we don't need this one either? :wink: